Assets/MixedRealityToolkit/Definitions/InputSystem/MixedRealityRaycastHit.cs

```diff
@@ -9,7 +9,7 @@ namespace Microsoft.MixedReality.Toolkit.Input
     /// <summary>
     /// The resulting hit information from an IMixedRealityRaycastProvider.
     /// </summary>
-    public struct MixedRealityRaycastHit
+    public class MixedRealityRaycastHit
```

Comment:
We were using a struct so that people could use MixedRealityRaycastHit in the same way as RaycastHit, and to make avoiding garbage easier.

If it's necessary to tell when a MixedRealityRaycastHit is 'valid' we could add a HitSomething field or a read-only Empty property that returns true if certain values haven't been set. Or we could just rely on the bool / out pattern that's already in place.

Reply:
Actually, @Railboy makes a very good point here @wiwei. These raycasts happen _very_ frequently and on the UI thread. If a Class is used instead of a Struct it's going to add _significant_ pressure on the garbage collector. If there is any way to use a struct instead here, it would be highly desirable.